dind-agent/Dockerfile

```diff
@@ -1,5 +1,6 @@
-FROM docker:1.9-dind
-RUN apk --update add openjdk7-jre unzip git python python3 bash jq openssh-client
+FROM docker:1.10-dind
+# include dependencies needed by the jenkins agent itself
```

Comment:
The absolute minimum packages required for this agent to launch and register with Jenkins would be:
* openjdk8-jre, to launch slave.jar
* bash, to launch wrapper.sh
* openssh-client, to populate the known_hosts file during the image build step

We provide the following packages for convenience, since they are probably used by most of our users:
* python and python3
* jq
* git
* unzip